### PR TITLE
fixes #14773 - add 'facts' to migration context

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

**Comment by domcleal** [comment]
[test] - one of the slaves was throwing errors since a security update, causing intermittent failures.

**Comment by mbacovsky** [comment]
:+1: Very useful, thanks @domcleal! Tested with foreman-installer on CentOS 7, works well.

File: test/kafo/migration_context_test.rb [hunk 1]
```diff
@@ -8,6 +8,22 @@ module Kafo
...
       specify { context.respond_to?(:answers) }
+      specify { context.respond_to?(:facts) }
+    end
+
+    describe '#facts' do
+      specify { MigrationContext.stub(:facts, {:foo => 'bar'}) { context.facts.must_equal(:foo => 'bar') } }
+    end
+
+    describe '.facts' do
+      specify { MigrationContext.stub(:`, {'foo' => 'bar'}.to_yaml) { MigrationContext.facts.must_equal(:foo => 'bar') } }
```
Comment (ares) [inline test/kafo/migration_context_test.rb]: nice trick :+1:

**Comment by ares** [comment]
Thanks @domcleal, merging!

**Pull request merged**